src/components/app/details/triggerView/cdMaterial.tsx

```diff
@@ -1,231 +1,844 @@
+import ReactSelect, { components } from 'react-select'
+import { CDMaterialProps, CDMaterialState, CDMaterialType, DeploymentWithConfigType } from './types'
+import { GitTriggers } from '../cIDetails/types'
+import close from '../../../../assets/icons/ic-close.svg'
+import arrow from '../../../../assets/icons/misc/arrow-chevron-down-black.svg'
+import { ReactComponent as Check } from '../../../../assets/icons/ic-check-circle.svg'
+import { ReactComponent as DeployIcon } from '../../../../assets/icons/ic-nav-rocket.svg'
+import { ReactComponent as WarningIcon } from '../../../../assets/icons/ic-warning.svg'
+import { ReactComponent as BackIcon } from '../../../../assets/icons/ic-arrow-backward.svg'
+import { ReactComponent as BotIcon } from '../../../../assets/icons/ic-bot.svg'
+import play from '../../../../assets/icons/misc/arrow-solid-right.svg'
+import docker from '../../../../assets/icons/misc/docker.svg'
+import {
+    VisibleModal,
+    ScanVulnerabilitiesTable,
+    Progressing,
+    getRandomColor,
+    showError,
+    shallowEqual,
+    ConditionalWrap,
+} from '../../../common'
+import { EmptyStateCdMaterial } from './EmptyStateCdMaterial'
+import { CDButtonLabelMap, getCommonConfigSelectStyles } from './config'
+import {
+    CDModalTab,
+    getLatestDeploymentConfig,
+    getRecentDeploymentConfig,
+    getSpecificDeploymentConfig,
+} from '../../service'
+import GitCommitInfoGeneric from '../../../common/GitCommitInfoGeneric'
+import { getModuleInfo } from '../../../v2/devtronStackManager/DevtronStackManager.service'
+import { ModuleNameMap } from '../../../../config'
+import { ModuleStatus } from '../../../v2/devtronStackManager/DevtronStackManager.type'
+import { DropdownIndicator, Option } from '../../../v2/common/ReactSelect.utils'
+import { getDeployConfigOptions, processResolvedPromise } from './TriggerView.utils'
+import TriggerViewConfigDiff from './triggerViewConfigDiff/TriggerViewConfigDiff'
+import Tippy from '@tippyjs/react'
 
 export class CDMaterial extends Component<CDMaterialProps, CDMaterialState> {
-  constructor(props: CDMaterialProps) {
-    super(props)
-    this.state = {
-      isSecurityModuleInstalled: false
-    }
-  }
-
-  componentDidMount() {
-    this.getSecurityModuleStatus()
-  }
-
-  async getSecurityModuleStatus(): Promise<void> {
-    try {
-        const { result } = await getModuleInfo(ModuleNameMap.SECURITY)
-        if (result?.status === ModuleStatus.INSTALLED) {
-          this.setState({ isSecurityModuleInstalled: true })
+    constructor(props: CDMaterialProps) {
+        super(props)
+        this.state = {
+            isSecurityModuleInstalled: false,
+            checkingDiff: false,
+            diffFound: false,
+            showConfigDiffView: false,
+            loadingMore: false,
+            showOlderImages: true,
+            noMoreImages: false,
+            selectedConfigToDeploy: {
+                label: 'Config deployed with selected image',
```

Comment:
Should we move the hardcoding to constants?

Reply:
Done. Created a const.